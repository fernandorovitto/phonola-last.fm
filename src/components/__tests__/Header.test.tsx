import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import Header from '../Header';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
describe('Test for Header Component', () => {
  test('Should render the component', () => {
    const wrapper = shallow(<Header title={'Super title'} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Should have the specific title', () => {
    const title = 'Test title';
    const wrapper = shallow(<Header title={title} />);
    expect(wrapper.contains(title)).toBe(true);
  });
});
