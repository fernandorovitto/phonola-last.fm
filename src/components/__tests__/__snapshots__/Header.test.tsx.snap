// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header Component Should render the component 1`] = `
<styled.header>
  <styled.div>
    <styled.div>
      <styled.h2>
        Super title
      </styled.h2>
    </styled.div>
    <styled.nav>
      <Styled(NavLink)
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Home
      </Styled(NavLink)>
    </styled.nav>
  </styled.div>
  <styled.div>
    <withRouter(Connect(SearchBar)) />
  </styled.div>
</styled.header>
`;
